src/amber/environment.cr

```diff
@@ -0,0 +1,40 @@
+require "./environment/**"
+require "./support/file_encryptor"
+
+module Amber::Environment
+  alias EnvType = String | Symbol
+
+  macro included
+    SECRET_KEY  = "AMBER_ENCRYPTION_KEY"
+    SECRET_FILE = "./.encryption_key"
+    CURRENT_ENVIRONMENT = ENV["AMBER_ENV"] ||= "development"
+    class_property environment_path : String = "./config/environments/"
+
+    @@_settings : Settings?
+
+    def self.settings
+      @@_settings ||= EnvironmentLoader.new(CURRENT_ENVIRONMENT, @@environment_path, file_encryptor).settings
+    end
+
+    def self.logger
+      settings.logger
+    end
+
+    def self.env=(env : EnvType)
+      ENV["AMBER_ENV"] = env.to_s
+      @@_settings = EnvironmentLoader.new(env, @@environment_path, file_encryptor).settings
+    end
+
+    def self.env
+      @@env ||= Environment::Environment.new(CURRENT_ENVIRONMENT)
+    end
+
+    def self.file_encryptor
+      @@file_encryptor ||= Support::FileEncryptor.new(encryption_key)
+    end
+
+    private def self.encryption_key
```

Comment:
Encryption key shouldn't be here since it's only used inside of FileEncryptor. Leaving encryption_key inside of FileEncryptor is more inline with SOLID principles.

Reply:
I agree and there was other part of the code using secret key. I think only 1 additional place

Reply:
If something else uses that it really shouldn't be. FileEncryptor should work in a vacuum and everything related to encrypted files should go through it.